### PR TITLE
Rohit/ceres 471 delete metric by metric group

### PR DESCRIPTION
**Pull request by rohitsngh27**
# Resolves
https://jira.rax.io/browse/CERES-471

# What
Add feature to delete metric by metric group.

# How
Added support to accept metric group parameter in delete metrics api request .

## How to test
This can be tested via unit tests or Rest api call.

### CODE REVIEW

**Review by shabd67 — APPROVED** [review]
looks good to me.

**Review by shabd67 — APPROVED** [review]

File: src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java [hunk 1]
```diff
@@ -46,23 +48,28 @@
...
     this.dataTablesStatements = dataTablesStatements;
     this.timeSlotPartitioner = timeSlotPartitioner;
     this.downsampleProperties = downsampleProperties;
+    this.appProperties = appProperties;
     this.metadataService = metadataService;
     this.metricDeletionHelper = metricDeletionHelper;
   }
 
-  public Mono<Empty> deleteMetrics(String tenant, String metricName, List<String> tag,
+  public Mono<Empty> deleteMetrics(String tenant, String metricName, String metricGroup, List<String> tag,
```
Comment (matsboman) [inline src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java:67]: We should separate this into its own method

File: src/main/java/com/rackspace/ceres/app/web/DeleteMetricController.java [hunk 2]
```diff
@@ -31,6 +32,7 @@ public DeleteMetricController(MetricDeletionService metricDeletionService) {
   @DeleteMapping
   public Mono<Empty> deleteMetrics(@RequestHeader(value = "#{appProperties.tenantHeader}") String tenantHeader,
       @RequestParam(required = false) String metricName,
+      @RequestParam(required = false) String metricGroup,
```
Comment (matsboman) [inline src/main/java/com/rackspace/ceres/app/web/DeleteMetricController.java:35]: I think again this should be separated out

**Review by matsboman — CHANGES_REQUESTED** [review]
I think we should think about if we could separate out things instead of doing If .. else if .. else if inside the methods.

File: src/main/java/com/rackspace/ceres/app/web/DeleteMetricController.java (hunk 2 above)
Comment (rohitsngh27) [reply src/main/java/com/rackspace/ceres/app/web/DeleteMetricController.java:35]: You mean like creating a new endpoint for deletion by metric group?

Comment (matsboman) [reply src/main/java/com/rackspace/ceres/app/web/DeleteMetricController.java:35]: yes

File: src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java [hunk 3]
```diff
@@ -136,7 +154,7 @@ public MetricDeletionService(DataTablesStatements dataTablesStatements,
    * @param end        the end
    * @return the mono
    */
-  private Mono<Empty> deleteMetricsByMetricNameAndTag(String tenant, String metricName,
+  public Mono<Empty> deleteMetricsByMetricNameAndTag(String tenant, String metricName,
       List<String> tag, Instant start, Instant end) {
     log.debug(
         "Deleting metrics {} with tag {} for tenant: {}  ",
```
Comment (matsboman) [inline src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java:160]: Put this on the same line

File: src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java [hunk 4]
```diff
@@ -106,7 +121,7 @@ public MetricDeletionService(DataTablesStatements dataTablesStatements,
    * @param end        the end
    * @return the mono
    */
-  private Mono<Empty> deleteMetricsByMetricName(String tenant, String metricName, Instant start,
+  public Mono<Empty> deleteMetricsByMetricName(String tenant, String metricName, Instant start,
       Instant end) {
     log.debug("Deleting metrics {} for tenant: {} ", metricName,
         tenant);
```
Comment (matsboman) [inline src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java:127]: Put on the same line

File: src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java [hunk 5]
```diff
@@ -119,7 +134,10 @@ public MetricDeletionService(DataTablesStatements dataTablesStatements,
               .then(deleteMetricsFromRawWithSeriesSetHash(tenant, startDateTime, end, seriesSetHash)))
               .then(Mono.empty());
     } else  {
-      start = Instant.now().minus(getHighestTTLGranularity().getTtl());
+      start = Instant.now().minus(
+          getHighestTTLGranularity().getTtl().toHours() + appProperties.getIngestStartTime()
+              .toHours(),
```
Comment (matsboman) [inline src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java:139]: remove line break

**Review by matsboman — APPROVED** [review]
Minor formatting suggestions otherwise LGTM

**Review by shabd67 — APPROVED** [review]

**Pull request merged**